src/solvers/fdsbp_unstructured/containers_2d.jl

```diff
@@ -0,0 +1,124 @@
+# !!! warning "Experimental implementation (curvilinear FDSBP)"
+#     This is an experimental feature and may change in future releases.
+
+# By default, Julia/LLVM does not use fused multiply-add operations (FMAs).
+# Since these FMAs can increase the performance of many numerical algorithms,
+# we need to opt-in explicitly.
+# See https://ranocha.de/blog/Optimizing_EC_Trixi for further details.
+@muladd begin
+
+# Alternative constructor to reuse the element container from `UnstructuredMesh2D`
+# OBS! New element constructors require the derivative operator to compute the metric terms.
+function init_elements!(elements::UnstructuredElementContainer2D, mesh, basis::AbstractDerivativeOperator)
+  four_corners = zeros(eltype(mesh.corners), 4, 2)
+
+  # loop through elements and call the correct constructor based on whether the element is curved
+  for element in eachelement(elements)
+    if mesh.element_is_curved[element]
+      init_element!(elements, element, basis, view(mesh.surface_curves, :, element))
+    else # straight sided element
+      for i in 1:4, j in 1:2
+        # pull the (x,y) values of these corners out of the global corners array
+        four_corners[i, j] = mesh.corners[j, mesh.element_node_ids[i, element]]
+      end
+      init_element!(elements, element, basis, four_corners)
+    end
+  end
+end
```

Comment:
Yes, that is the only difference. I think it can be unified with some minor changes in the DGSEM version. First I want to work out how to get this to work for SBP or Upwind.